### PR TITLE
Optimizations for the generated parser

### PR DESCRIPTION
**Pull request by austin-schick**
### Changes made

- Use a while loop and `break` to simulate the C parser's `goto` behavior, instead of creating and calling a `done()` function
-  Stop running `PyBytes_FromStringAndSize` when tokenizing. I verified that `.bytes` was never read during execution. But it's possible that this wasn't a valid change.
- Add the nodejs profiling script to source control
- Small improvements to scripts and tests

### Performance improvements

Here's are the times I see on my laptop for the datetime tests:

```
First PEG parser:                      193ms
Generated PEG parser (before this PR): 130ms
Generated PEG parser (after this PR):   88ms
Legacy parser:                          55ms
```

And here's an additional datapoint from my pyperformance tests:

```
First PEG parser:                      2.8s
Generated PEG parser (before this PR): 2.6s
Generated PEG parser (after this PR):  2.1s
Legacy parser:                         1.2s
```

So I think we're sitting around 1.7x slower than the legacy parser after these improvements.

### CODE REVIEW

File: www/src/pegen.js [hunk 1]
```diff
@@ -15,6 +15,11 @@
 var _b_ = __BRYTHON__.builtins
 
 const Load = new $B.ast.Load()
+const NULL = undefined;
+const ENDMARKER = 0,
```
Comment (austin-schick) [inline www/src/pegen.js:19]: This was necessary to get the nodejs profiler running, but didn't seem necessary in the browser. I didn't have the time to dig into exactly why that discrepancy exists.

**Comment by PierreQuentel** [comment]
Once again impressive work Austin !

I am currently working on a separate branch `generated_peg_parser` and making good progress to pass all the tests in the built-in test suite. I am not a GIT expert, is it possible to apply this PR to the branch ?

**Comment by austin-schick** [comment]
Thank you, and yes! I updated this PR to merge into the `generated_peg_parser` branch. It looks like there are a few merge conflicts -- I should be able to look into those this week.

**Comment by austin-schick** [comment]
Merge conflicts fixed! I verified that datetime still parses, but I didn't run the full test suite on this branch.

**Comment by PierreQuentel** [comment]
I ran the test suite with the page **`/tests/parse_tests/index_generated_parser.html`**.

After fixing a bug in **`pegen.js`**, I have merged your PR. The whole test suite passes, slightly slower than with the legacy parser, but the difference is negligible compared to the benefits of the new parser.

The next step is to test all the scripts in the gallery, the documentation etc. before I can merge the branch into master.

Thanks once again Austin, this is a major step forward for Brython !

**Comment by austin-schick** [comment]
Very exciting! I'm happy to have been able to help out some. Thanks for all your work on this Pierre!

**Pull request merged**